gammapy/maps/core.py

```diff
@@ -108,7 +108,12 @@ def data(self, value):
             raise TypeError("Map data must be a Numpy array. Set unit separately")
 
         if not value.shape == self.geom.data_shape:
-            value = value.reshape(self.geom.data_shape)
+            try:
```

Comment:
this can be dangerous. Because you are catching all the exceptions. Let's say that there is an error which is caused by something other than the shape. Then you will have a cryptic error message about the shape and not the correct error message.

Reply:
modified to raise a `ValueError` after a `ValueError`